packages/eslint-plugin/src/rules/prefer-includes.ts

```diff
@@ -155,6 +155,21 @@ export default createRule({
           }
         }
 
+        // Check if `indexOf` is used as part of `includes` method declaration
+        let parent = node.parent;
+        while (parent) {
+          if (
+            (parent.type === AST_NODE_TYPES.MethodDefinition &&
+              parent.key.type === 'Identifier' &&
+              parent.key.name === 'includes') ||
+            // to handle a case where a class is declared inside `includes` method
+            parent.type === AST_NODE_TYPES.ClassDeclaration
```

Comment:
I don't think that this check (`=== "ClassDeclaration"`) is proper. All `xxx.indexOf()` in classes will be ignored.